src/server/object_services/map_deleter.js

```diff
@@ -43,6 +44,18 @@ async function delete_chunks_if_unreferenced(chunk_ids) {
     }
 }
 
+/**
+ * @param  {nb.ObjectMD} object
+*/
+async function delete_empty_object(object) {
+    if (!object) return;
+    dbg.log2('delete_empty_objects: object_ids', object);
+    const part_ids = await MDStore.instance().find_all_parts_of_object(object);
```

Comment:
This is doing a lot more work than we need - we only care if there exist any parts, and we don't care how many or to return their information. So we can do a findOne type of query (limit results to 1), but in any case I would set the MDStore API to only ask if `has_any_parts_for_object()` similar to all the other `has_any_*` functions there.